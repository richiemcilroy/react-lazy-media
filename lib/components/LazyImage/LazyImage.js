"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LazyImage = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _ReactLazyMedia = require("../../ReactLazyMedia");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj["default"] = obj; return newObj; } }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _nonIterableSpread(); }

function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance"); }

function _iterableToArray(iter) { if (Symbol.iterator in Object(iter) || Object.prototype.toString.call(iter) === "[object Arguments]") return Array.from(iter); }

function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = new Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var LazyImage =
/*#__PURE__*/
function (_Component) {
  _inherits(LazyImage, _Component);

  function LazyImage() {
    _classCallCheck(this, LazyImage);

    return _possibleConstructorReturn(this, _getPrototypeOf(LazyImage).apply(this, arguments));
  }

  _createClass(LazyImage, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      _ReactLazyMedia.ReactLazyMedia.getInstance();
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          alt = _this$props.alt,
          classes = _this$props.classes,
          id = _this$props.id,
          lqip = _this$props.lqip,
          src = _this$props.src,
          width = _this$props.width,
          height = _this$props.height,
          onError = _this$props.onError;
      return _react["default"].createElement("img", {
        alt: alt,
        className: _classnames["default"].apply(void 0, ['lazy'].concat(_toConsumableArray(classes))),
        src: lqip,
        "data-src": src,
        height: height,
        id: id,
        width: width,
        onError: onError
      });
    }
  }]);

  return LazyImage;
}(_react.Component);

exports.LazyImage = LazyImage;

_defineProperty(LazyImage, "propTypes", {
  alt: _propTypes["default"].string,
  classes: _propTypes["default"].arrayOf(_propTypes["default"].string),
  id: _propTypes["default"].string,
  src: _propTypes["default"].string.isRequired,
  width: _propTypes["default"].number,
  height: _propTypes["default"].number,
  lqip: _propTypes["default"].string,
  // Low-quality image placeholder
  onError: _propTypes["default"].func
});

_defineProperty(LazyImage, "defaultProps", {
  classes: []
});