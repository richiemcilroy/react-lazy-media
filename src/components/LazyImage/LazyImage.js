import React, {Component} from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import ReactLazyload from '../../ReactLazyload';

export class LazyImage extends Component {
  static propTypes = {
    alt: PropTypes.string,
    classes: PropTypes.arrayOf(PropTypes.string),
    id: PropTypes.string,
    src: PropTypes.string.isRequired,
    width: PropTypes.number,
    height: PropTypes.number,
    lqip: PropTypes.string // Low-quality image placeholder
  };

  static defaultProps = {
    classes: []
  };

  // Update lazyLoad after first rendering of every image
  componentDidMount() {
    ReactLazyload.getInstance().update();
  }

  // Update lazyLoad after re-rendering of every image
  componentDidUpdate() {
    ReactLazyload.getInstance().update();
  }

  render() {
    const {alt, classes, id, lqip, src, width, height} = this.props;
    return (
      <img
        alt={alt}
        className={classNames('lazy', ...classes)}
        src={lqip}
        data-src={src}
        height={height}
        id={id}
        width={width}
      />
    );
  }
}

export default LazyImage;
