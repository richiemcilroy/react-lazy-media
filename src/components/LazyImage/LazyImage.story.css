.lazyImage {
  width: 200px;
  height: 300px;
  margin-top: 150vh;
}
