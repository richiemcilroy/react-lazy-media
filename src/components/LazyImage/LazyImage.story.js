import React from 'react';
import {storiesOf} from '@storybook/react';
import {withKnobs, text} from '@storybook/addon-knobs';
import LazyImage from './LazyImage';
import './LazyImage.story.css';

storiesOf('Lazy', module)
  .addDecorator(withKnobs)
  .add('LazyImage', () => {
    return <LazyImage classes={['lazyImage']} src={'http://placekitten.com/200/300'} />;
  });
