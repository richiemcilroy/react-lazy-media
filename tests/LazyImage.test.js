import React from 'react';
import renderer from 'react-test-renderer';
import {LazyImage} from '../src/components/LazyImage/LazyImage';

test('LazyImage should require a src prop', () => {
  expect(() => {
    renderer.create(<LazyImage />);
  }).toThrowErrorMatchingSnapshot();
});

test('LazyImage should have default props', () => {
  const component = renderer.create(<LazyImage src={'/image.jpeg'} />);

  const tree = component.toJSON();

  expect(tree.props).toEqual({
    'data-src': '/image.jpeg',
    className: 'lazy'
  });
});

test('LazyImage should render an image with a src', () => {
  const component = renderer.create(<LazyImage src={'/image.jpeg'} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('LazyImage should render an image with a src and alt text', () => {
  const component = renderer.create(<LazyImage src={'/image.jpeg'} alt={'My image'} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('LazyImage render an image with a low quality image placeholder (lqip)', () => {
  const component = renderer.create(<LazyImage src={'/image.jpeg'} lqip={'/lqip.jpeg'} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('LazyImage should have 300px width', () => {
  const component = renderer.create(<LazyImage src={'/image.jpeg'} width={300} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('LazyImage should have 300px height', () => {
  const component = renderer.create(<LazyImage src={'/image.jpeg'} height={300} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('LazyImage should have an id', () => {
  const component = renderer.create(<LazyImage src={'/image.jpeg'} id={'lazyImage'} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('LazyImage should have classes', () => {
  const component = renderer.create(<LazyImage src={'/image.jpeg'} classes={['class1', 'class2']} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('LazyImage should have onError', () => {
  const component = renderer.create(<LazyImage src={'/image.jpeg'} onError={console.log('Error!')} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
