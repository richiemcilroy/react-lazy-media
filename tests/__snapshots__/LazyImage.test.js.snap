// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyImage render an image with a low quality image placeholder (lqip) 1`] = `
<img
  className="lazy"
  data-src="/image.jpeg"
  src="/lqip.jpeg"
/>
`;

exports[`LazyImage should have 300px height 1`] = `
<img
  className="lazy"
  data-src="/image.jpeg"
  height={300}
/>
`;

exports[`LazyImage should have 300px width 1`] = `
<img
  className="lazy"
  data-src="/image.jpeg"
  width={300}
/>
`;

exports[`LazyImage should have an id 1`] = `
<img
  className="lazy"
  data-src="/image.jpeg"
  id="lazyImage"
/>
`;

exports[`LazyImage should have classes 1`] = `
<img
  className="lazy class1 class2"
  data-src="/image.jpeg"
/>
`;

exports[`LazyImage should have onError 1`] = `
<img
  className="lazy"
  data-src="/image.jpeg"
/>
`;

exports[`LazyImage should render an image with a src 1`] = `
<img
  className="lazy"
  data-src="/image.jpeg"
/>
`;

exports[`LazyImage should render an image with a src and alt text 1`] = `
<img
  alt="My image"
  className="lazy"
  data-src="/image.jpeg"
/>
`;

exports[`LazyImage should require a src prop 1`] = `
"Warning: Failed prop type: The prop \`src\` is marked as required in \`LazyImage\`, but its value is \`undefined\`.
    in LazyImage"
`;
